agent.py

```diff
@@ -361,57 +357,51 @@ def doReproduction(self):
         # Agent marked for removal or not interested in reproduction should not reproduce
         if self.alive == False or self.isFertile() == False:
             return
-        neighborCells = self.cell.neighbors
-        random.shuffle(neighborCells)
+        random.shuffle(self.adjacentNeighbors)
         emptyCells = self.findEmptyNeighborCells()
```

Comment:
Everything changed below is just indentation differences from a removed if statement.